test/spec/modules/prebidServerBidAdapter_spec.js

```diff
@@ -2910,6 +2910,25 @@ describe('S2S Adapter', function () {
       expect(requestBid.coopSync).to.be.undefined;
     });
 
+    it('should set imp banner if ortb2Imp.banner is present', function() {
+      const consentConfig = { s2sConfig: CONFIG };
+      config.setConfig(consentConfig);
+      const bidRequest = utils.deepClone(REQUEST);
+      bidRequest.ad_units[0].ortb2Imp = {
+        banner: {
+          api: 7
+        }
+      };
+
+      adapter.callBids(bidRequest, BID_REQUESTS, addBidResponse, done, ajax);
+      const parsedRequestBody = JSON.parse(server.requests[0].requestBody);
+
+      expect(parsedRequestBody.imp).to.be.a('array');
+      expect(parsedRequestBody.imp[0]).to.be.a('object');
+      expect(parsedRequestBody.imp[0]).to.have.property('banner');
+      expect(parsedRequestBody.imp[0].banner.api).to.equal(7);
```

Comment:
IMO only the last assertion is needed. #2 and #3 have to be true for #4 to be true. #1 is better suited for other tests.

Reply:
Funny how GitHub linked those prs